core/lib/rom/plugins/schema/timestamps.rb

```diff
@@ -0,0 +1,58 @@
+module ROM
+  module Plugins
+    module Schema
+      # A plugin for automatically adding timestamp fields
+      # to the schema definition
+      #
+      # @example
+      #   schema do
+      #     use :timestamps
+      #   end
+      #
+      #   # using non-default names
+      #   schema do
+      #     use :timestamps, attributes: %i(created_on updated_on)
+      #   end
+      #
+      #   # using other types
+      #   schema do
+      #     use :timestamps, type: Types::Date
+      #   end
+      #
+      # @api public
+      module Timestamps
+        DEFAULT_TIMESTAMPS = %i(created_at updated_at).freeze
+
+        # @api private
+        def self.apply(schema, options)
+          type = options.fetch(:type, Types::Time)
+          names = options.fetch(:attributes, DEFAULT_TIMESTAMPS)
+          attributes = names.map { |name| type.meta(name: name, source: schema.name) }
+
+          schema.attributes.concat(
+            schema.class.attributes(attributes, schema.attr_class)
+          )
```

Comment:
yeah, this is what we need, add attributes to an existing schema before it gets frozen but without using the dsl